internal/ingest/goa_batch.go

```diff
@@ -13,7 +18,69 @@ func (svc *ingestImpl) AddBatch(
 	ctx context.Context,
 	payload *goaingest.AddBatchPayload,
 ) (*goaingest.AddBatchResult, error) {
-	return nil, ErrNotImplemented
+	if payload == nil {
+		return nil, goaingest.MakeNotValid(errors.New("missing payload"))
+	}
+
+	sourceID, err := uuid.Parse(payload.SourceID)
+	if err != nil {
+		return nil, goaingest.MakeNotValid(errors.New("invalid SourceID"))
+	}
+
+	if len(payload.Keys) == 0 {
+		return nil, goaingest.MakeNotValid(errors.New("empty Keys"))
+	}
+
+	claims, err := checkClaims(ctx)
+	if err != nil {
+		return nil, goaingest.MakeNotValid(err)
+	}
+
+	// TODO: Discuss identifier generation strategy.
+	bUUID := uuid.Must(uuid.NewRandomFromReader(svc.rander))
+	identifier := fmt.Sprintf("Batch-%s", bUUID.String())
+	if payload.Identifier != nil && *payload.Identifier != "" {
+		identifier = *payload.Identifier
+	}
+	b := &datatypes.Batch{
+		UUID:       bUUID,
+		Status:     enums.BatchStatusQueued,
+		Identifier: identifier,
+		SIPSCount:  len(payload.Keys),
+	}
+
+	if claims != nil {
+		b.Uploader = &datatypes.User{
```

Comment:
Isn't this generating a new user for every batch created? I think we want to link to an existing user when possible.